packages/components/addon/components/hds/accordion/item/button.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+export default class HdsAccordionItemButtonComponent extends Component {
+  /**
+   * Get the class names to apply to the component.
+   * @method ItemButton#classNames
+   * @return {string} The "class" attribute to apply to the component.
+   */
+  get classNames() {
+    let classes = ['hds-accordion-item__button'];
+
+    // add a class based on the @isOpen argument
+    if (this.args.isOpen) {
+      classes.push('hds-accordion-item__button--is-open');
+    }
+
+    // if toggle is NOT interactive, then button IS
+    if (this.args.isInteractive === false) {
```

Comment:
I find this confusing. We're testing against an exact value (`false`) which means `true` and `undefined` fall under the same category, which is weird.

What about having a default value (that I imagine is the `true` condition, given that it's `true` in the parent element) via a getter, and avoid this strange conditional?

Something like this:
```
get parentContainsInteractive() {
    return this.args.parentContainsInteractive ?? [true|false]; // depends on how you define the default in the parent
}
```
and then

```
if (this.parentContainsInteractive) {
    classes.push('hds-accordion-item__button--parent-contains-interactive');
} else {
    classes.push('hds-accordion-item__button--parent-does-not-contain-interactive');
}
```

Reply:
I'm fixing the issue with the logic but I don't think it needs a getter as we're setting a default in the AccordionItem parent. (See what you think after I push up the changes.)